proposals/0245-sharing-wifi-ssid-and-password.md

```diff
@@ -18,627 +18,1033 @@ If WiFi credentials are automatically generated by the IVI system, the password
 
-1. The policy table needs to be updated for the new Mobile API.
-2. The new functional group should be created so that, if required, the user can explicitly disable the sharing of WiFi info.
+<!--
+@startuml ../assets/proposals/0245-sharing-wifi-ssid-and-password/vehicle_access_point
+!include ../assets/proposals/0245-sharing-wifi-ssid-and-password/vehicle_access_point.puml
+@enduml
+-->
+![Vehicle acess point](../assets/proposals/0245-sharing-wifi-ssid-and-password/vehicle_access_point.png)
 
-### Proxy Changes
+<!--
+@startuml ../assets/proposals/0245-sharing-wifi-ssid-and-password/mobile_access_point
+!include ../assets/proposals/0245-sharing-wifi-ssid-and-password/mobile_access_point.puml
+@enduml
+-->
+![Mobile acess point](../assets/proposals/0245-sharing-wifi-ssid-and-password/mobile_access_point.png)
 
-1. The proxy needs to implement the Mobile API mentioned above.
-2. The mobile device and IVI can connect via WiFi using platform APIs provided SSID, password, and security type are known. However, these APIs need special permissions or entitlements. App developers can choose if they want to provide these permissions or not. Hence, app developers should be able to disable WiFi auto-connect feature using proxy API. The proxy will provide an API to app developers to enable or disable WiFi auto-connect features. If app developers use this API to enable WiFi auto-connection features, the proxy should use the below mentioned RPCs to request WiFi credentials or send WiFi credentials to the IVI. If app developers use this API to disable WiFi auto-connection, the proxy should not request or provide WiFi credentials to the IVI.
+<!--
+@startuml ../assets/proposals/0245-sharing-wifi-ssid-and-password/vehicle_access_point_multiapps
+!include ../assets/proposals/0245-sharing-wifi-ssid-and-password/vehicle_access_point_multiapps.puml
+@enduml
+-->
+![Vehicle access point multiple apps](../assets/proposals/0245-sharing-wifi-ssid-and-password/vehicle_access_point_multiapps.png)
 
-Here is an example of the WiFi configuration APIs to set WiFi configuration in Android proxy:
+### Multiple device connection
 
-```java
-/** API to make mobile device as WiFi access point*/
-MultiplexTransportConfig.setCanHostWiFiNetwork(Boolean);
+#### Vehicle is a network host
 
-/** API to automatically connect WiFi access point created by IVI*/
-MultiplexTransportConfig.setCanAutoJoinWiFiNetwork(Boolean);
-```
+For each connected device, SDL Core will select a single feature enabled application.
+
+Once network parameters are established with the first device, `OnNetworkConfiguration` will be sent to this application and HMI. After receiving this notification, HMI should send `OnSystemCapabilityUpdated` in case capabilities have changed. The next application's capabilities will be compared to updated HMI capabilities.

```

Comment:
`OnNetworkConfiguration` should be `OnSystemCapabilityUpdated`

Reply:
Fixed ab0e7eb